### DISCUSSION TITLE
Test Automation University Contract tests with Pact course getting errors.

### DISCUSSION THREAD

**Discussion by druoid** [pact-foundation/pact-js#1082] [Q&A]
When I run npm run test:consumer I get the following error.  When I remove line 26 the tests passes.

    Pact verification failed - expected interactions did not match actual.

      24 |    }]
      25 | 
    > 26 |    afterEach(() => provider.verify())
         |                             ^
      27 | 
      28 |    describe("GET Clients", () => {
      29 |      beforeEach(() => {

      at new VerificationError (node_modules/@pact-foundation/pact/src/errors/verificationError.js:21:42)
      at Pact.verify (node_modules/@pact-foundation/src/httpPact/index.ts:215:13)
      at Object.<anonymous> (__tests__/contract/consumer/ClientsConsumer.spec.js:26:29)
      at processTicksAndRejections (node:internal/process/task_queues:95:5)

    
      Mock server failed with the following mismatches:
    
        0) The following request was expected but not received: 
            Method: GET
            Path: /clients
            Headers:
              Accept: application/json, text/plain, */*

Test Suites: 1 failed, 1 total
Tests:       1 failed, 1 total
Snapshots:   0 total
Time:        0.855s, estimated 1s
Ran all test suites matching /__tests__\/contract\/consumer/i.

**Comment by mefellows** [comment]
Can you please share the code? This usually means a mishandled promise or that your API client is not sending the request to the mock server

**Reply by druoid** [reply]
Thanks @mefellows all code can be found here https://github.com/druoid/pactpoc/tree/main/tau-pact-nodejs-course-1.0.1

**Comment by mefellows** [comment]
The problem is this line is not sending a request to the pact mock service, it's sending it to your provider. You don't start your provider in a consumer test, the whole point is to decouple them:

```
const response = await getClients()
```

Check out the examples in this repository for how to do this.

**Reply by druoid** [reply]
Thanks for your response @mefellows.  Unfortunately I couldn't get it working so have moved on to another example using pact.

**Reply by mefellows** [reply]
No worries. We have many you can borrow.

Might be worth raising with the original author that the examples don't run.
